public/js/common/grid_image.js

```diff
@@ -8,12 +8,12 @@ function getGridImage(session) {
     session = session.charAt(0).toUpperCase() + session.slice(1);
     var courses = getCoursesTable(session);
     $.ajax({
-        url: 'timetable-image',
-        data: {courses: courses, session: session},
+        url: 'calendar',
+        data: {session: session},
         success: function (data) {
             var contentDiv = $('<div></div>');
             var topContentDiv = $('<div></div>');
-            var calendarOption = $('<a href="calendar">Download ICS</a>');
+            var calendarOption = $('<a href="timetable-pdf" onclick="getPDF()">Download ICS</a>');
```

Comment:
You don't need a new AJAX request here (no `onclick` necessary). You can add a `target="_blank"` to open the PDF in a new tab.